ledgercomm/interfaces/hid_device.py

```diff
@@ -53,13 +60,44 @@ def open(self) -> None:
         """
         if not self.__opened:
             if not self.path:
-                self.path = HID.enumerate_devices(self.vendor_id)[0]
+                self.path = HID._decide_device_path(self.vendor_id)
             self.device.open_path(self.path)
             self.device.set_nonblocking(True)
             self.__opened = True
 
     @staticmethod
-    def enumerate_devices(vendor_id: int = 0x2C97) -> List[bytes]:
+    def _decide_device_path(vendor_id: int = DEFAULT_VENDOR_ID) -> bytes:
+        if hid is None:
+            raise HIDAPINotInstalledError()
+
+        devices: List[Mapping[str, Any]] = list(hid.enumerate(vendor_id, 0))
+        LOG.debug("hid.enumerate(), devices: %s", devices)
+
+        devices = [device for device in devices if device.get("interface_number") == 0]
+        if len(devices) == 1:
+            # No ambiguity, pick the only device
+            return devices[0]["path"]
+
+        # On MacOS, "interface_number" is not a reliable filtering criteria, so we
+        # also filter by "usage_page".
+        devices = [device for device in devices if device.get(
+            "usage_page") == MACOS_USAGE_PAGE]
+        if len(devices) == 1:
+            # No ambiguity, pick the only device
+            return devices[0]["path"]
+
+        if len(devices) > 1:
+            LOG.warn(
+                "More than one Ledger device with vendor_id %s, will pick the first one", hex(vendor_id))
+
+            # First, we sort by "path", so that the order is deterministic
+            devices = sorted(devices, key=lambda device: device["path"])
+            return devices[0]["path"]
+
+        raise Exception(f"Can't find Ledger device with vendor_id {hex(vendor_id)}")
+
+    @staticmethod
+    def enumerate_devices(vendor_id: int = DEFAULT_VENDOR_ID) -> List[bytes]:
```

Comment:
The logic of this function is left untouched - in case downstream applications were actually relying on the results being a collection of path, instead of a single path.